pkg/devfile/adapters/kubernetes/utils/utils.go

```diff
@@ -361,21 +362,32 @@ func PipeStdOutput(cmdOutput string, reader *io.PipeReader, controlC chan os.Sig
 	}
 }
 
-func CreateDockerConfigDataFromFilepath(DockerConfigJSONFilename string) ([]byte, error) {
+func CreateDockerConfigDataFromFilepath(DockerConfigJSONFilename string, fs afero.Fs) ([]byte, error) {
+	var filePointer io.Reader
 	filename, err := homedir.Expand(DockerConfigJSONFilename)
 	if err != nil {
 		return nil, fmt.Errorf("failed to generate path to file for %s: %v", DockerConfigJSONFilename, err)
 	}
-
-	f, err := os.Open(filename)
-	if err != nil {
-		return nil, fmt.Errorf("failed to read Docker config %#v : %s", filename, err)
+	if fs != nil {
+		f, err := fs.Open(filename)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read test Docker config %#v : %s", filename, err)
+		}
+		filePointer = f
+		defer f.Close()
+	} else {
```

Comment:
Why introduce the complexity here to this rather simple straightforward code?

Reply:
I was trying to write test cases for this function using an afero FS object but had no way to pass the test filesystem object to the function so I had to restructure it to accept the filesystem object so I could run the test

Reply:
Removed use of afero fs, just using os now